src/hidapi/SDL_hidapi.c

```diff
@@ -889,7 +899,8 @@ static const struct {
     Uint16 vendor;
     Uint16 product;
 } SDL_libusb_whitelist[] = {
-    { 0x057e, 0x0337 } // Nintendo WUP-028, Wii U/Switch GameCube Adapter
+    { 0x057e, 0x0337 }, // Nintendo WUP-028, Wii U/Switch GameCube Adapter
+    { 0x057e, 0x2073 }, // Nintendo Switch 2 NSO GameCube Controller
```

Comment:
Can we use USB_VENDOR_NINTENDO, USB_PRODUCT_NINTENDO_SWITCH2_GAMECUBE_CONTROLLER and  USB_PRODUCT_NINTENDO_GAMECUBE_ADAPTER here? This is more a question for @slouken

Reply:
Done!